src/Microsoft.AspNet.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcCoreBuilderExtensions.cs

```diff
@@ -133,12 +133,10 @@ internal static void AddViewServices(IServiceCollection services)
             //
             // Temp Data
             //
-            // Holds per-request data so it should be scoped
-            services.TryAddScoped<ITempDataDictionary, TempDataDictionary>();
-            services.TryAddScoped<SaveTempDataFilter>();
-
             // This does caching so it should stay singleton
             services.TryAddSingleton<ITempDataProvider, SessionStateTempDataProvider>();
+            services.TryAddSingleton<ITempDataDictionaryFactory, TempDataDictionaryFactory>();
+            services.TryAddSingleton<SaveTempDataFilter>();
```

Comment:
Comment reads like it might apply to these two lines. Suggest an additional blank line to help. Or add a comment saying these two depend on `ITempDataProvider` directly or indirectly.